yt_dlp/extractor/smotrim.py

```diff
@@ -0,0 +1,155 @@
+            }
+        }, {  # article (geo-restricted)
+            'only_matching': True,
+            'url': 'https://smotrim.ru/article/2813445',
+            'md5': 'e0ac453952afbc6a2742e850b4dc8e77',
+            'info_dict': {
+                'id': '2431846',
+                'ext': 'mp4',
+                'title': 'Новости культуры. Съёмки первой программы "Большие и маленькие"',
+                'description': 'md5:94a4a22472da4252bf5587a4ee441b99',
+            }
+        }, {  # brand, redirect
+            'only_matching': True,
+            'url': 'https://smotrim.ru/brand/64356',
+            'md5': '740472999ccff81d7f6df79cecd91c18',
+            'info_dict': {
+                'id': '2354523',
+                'ext': 'mp4',
+                'title': 'Большие и маленькие. Лучшее. 4-й выпуск',
+                'description': 'md5:84089e834429008371ea41ea3507b989',
+            }
+        }, {  # w/o video
+            'only_matching': True,
+            'url': 'https://smotrim.ru/article/2909569',
+        }
+    ]
+
+    def _real_extract(self, url):
+        webpage = self._download_webpage(url, None, 'Downloading webpage')
+        if re.match(self._VALID_URL, url).group('type') == 'brand':
+            video_id = self._search_regex(
+                r'"https://player.smotrim.ru/iframe/video/id/(?P<video_id>[0-9]+)/',
+                webpage, 'video_id', default=None)
+            if video_id is None:
+                raise ExtractorError('This page doesn\'t contain video.', expected=True)
+            webpage = self._download_webpage('https://smotrim.ru/video/' + video_id, None, 'Redirect to video')
+
+        # Example: https://player.smotrim.ru/iframe/video/id/1539617/start_zoom/true/showZoomBtn/false/sid/smotrim/isPlay/true/mute/true/?acc_video_id=1488382"
+        m = re.search(
+            r'"https://player.smotrim.ru/iframe/video/id/(?P<video_id>[0-9]+)/[^"]*?/\?acc_video_id=(?P<player_id>[0-9]+)"',
+            webpage)
+        if m is None:
+            raise ExtractorError('This page doesn\'t contain video.', expected=True)
+        video_id = m.group('video_id')
+        player_id = m.group('player_id')
+        player_id_split = '/'.join(re.findall('...', player_id.zfill(9)))  # 2624356 > 002/624/356
+
+        meta = self._download_json(f'https://player.smotrim.ru/iframe/datavideo/id/{video_id}/sid/smotrim', None,)
+        meta = traverse_obj(meta, ('data', 'playlist', 'medialist'))[0]
+
+        video_quality = {
+            '234': 'low-wide',
+            '336': 'low',
+            '360': 'medium-wide',
+            '528': 'high',
+            '540': 'high-wide',
+            '720': 'hd-wide',
+            '1080': 'fhd-wide',
+        }
+        video_template = 'https://cdn-v.rtr-vesti.ru/_cdn_auth/secure/v/vh/mp4/{quality}/{id_split}.mp4?auth=mh&vid={id}'
+
+        formats = []
+        if meta.get('errors') == '':
+            for key in traverse_obj(meta, ('sources', 'http')):
```

Comment:
again, despite traverse_obj, the code is actually fatal. Decide which one it should be